crates/sui-graphql-rpc/src/error.rs

```diff
@@ -74,8 +74,6 @@ pub enum Error {
     DomainParse(#[from] DomainParseError),
     #[error(transparent)]
     DbValidation(#[from] DbValidationError),
-    #[error("Provide one of digest or sequence_number, not both")]
```

Comment:
Instead of treating like this like an error, for consistency and simplicity we just apply both filters, and if there is not a checkpoint that matches both criteria (has that digest and that sequence number) an empty result is created.  This matches how `TransactionFilter`, `ObjectFilter`, etc would work.